core/modules/catalog/helpers/slugifyCategories.ts

```diff
@@ -20,11 +20,13 @@ const createSlug = (category: ChildrenData): string => {
 
 const slugifyCategories = (category: Category | ChildrenData): Category | ChildrenData => {
   if (category.children_data) {
-    for (let subcat of category.children_data) {
-      if (subcat.name) {
-        return slugifyCategories({ ...subcat, slug: createSlug(subcat) } as any as ChildrenData)
+    category.children_data.forEach((subCat: ChildrenData): void => {
```

Comment:
@1070rik What about shomething like this?
```
category.children_data = category.children_data.map(ch => ({
  ...ch,
  ...(ch.name ? { slug: createSlug(subCat) } : {})
}))
```

Reply:
Although I don't really see the added value of doing this instead of a forEach, this would only work for the first 2 levels of categories.
The `slugifyCategories(subCat)` call at the end of the loop would cause a recursive loop which would give the subcategories of that subcategory a slug.

Reply:
@1070rik yes, but I didn't mention about deleting `slugifyCategories(subCat)` my changes are related to only `for`, it's just simpler, and actually entire function could be refactored with using `map` call. However - it's your choice, just let me know if you want to spare time on it or not - it's still good PR (it solves the issue,  thanks!)

Reply:
Turning it into a map and doing the function call doesn't really make the code look better or easier, it only makes it more complex in my opinion. I still prefer the forEach method over the map since it easier to understand.

Reply:
for me, it looks much better, it's just imperative vs functional. In this case, it maybe doesn't change so much, but remember that in the `for` you are mutating values while `map` returns a new array - no side effects (they could be painful sometimes). As I mentioned entire function could be refactored using `map` - instead of modifying category you can return a new one with slugs inside (again, no side effects). 

Of course, this solution is also fine, thanks!